File: Programming-languages-and-laboratory/Java/Lessons/34_20221221/29-soluzione-a/Java/Dipendente.java
import java.util.*;

public abstract class Dipendente {
	
	protected int codice;
	protected String  nome;
	protected double costo;
	protected String tipo;

	
	public Dipendente(int codice,  String nome, String tipo,  double costo){
    	this.codice=codice;
		this.nome=nome;
		this.costo=costo;
		this.tipo=tipo;
		}
	public double getCosto()
	{
		return costo;
	}
		public int getCod()
	{
		return codice;
	}
	public String getNome()
	{
		return nome;
	}
	public String toString(){
		return nome+"\t"+codice+"\t"+tipo;
	}
}

File: Programming-languages-and-laboratory/Python/Lessons/34_20221221/Dipendente.py
class Dipendente:
    def __init__(self, codice, tipo, nome, costo_orario):
        self._codice = codice
        self._tipo = tipo
        self._nome = nome
        self._costo_orario = costo_orario
        self._numero_servizi = 0

    def get_numero_servizi(self):
        return self._numero_servizi

    def add_servizio(self):
        self._numero_servizi += 1

    def get_costo(self):
        return self._costo_orario

    def get_nome(self):
        return self._nome

    def get_codice(self):
        return self._codice

    def __str__(self):
        return self._nome
